tests/e2e/service_invocation/service_invocation_test.go

```diff
@@ -89,27 +97,91 @@ var serviceinvocationTests = []struct {
 	},
 }
 
+var moreServiceinvocationTests = []struct {
+	in               string
+	remoteApp        string
+	appMethod        string
+	expectedResponse string
+}{
+	// For descriptions, see corresponding methods in dapr/tests/apps/service_invocation/app.go
+	{
+		"Test HTTP to HTTP",
+		"serviceinvocation-callee-1",
+		"httptohttptest",
+		"success",
+	},
+	{
+		"Test HTTP to gRPC",
+		"grpcapp",
+		"httptogrpctest",
+		"success",
+	},
+	{
+		"Test gRPC to HTTP",
+		"serviceinvocation-callee-1",
+		"grpctohttptest",
+		"success",
+	},
+	{
+		"Test gRPC to gRPC",
+		"grpcapp",
+		"grpctogrpctest",
+		"success",
+	},
+}
+
 func TestServiceInvocation(t *testing.T) {
 	externalURL := tr.Platform.AcquireAppExternalURL("serviceinvocation-caller")
 	require.NotEmpty(t, externalURL, "external URL must not be empty!")
-
+	var err error
 	// This initial probe makes the test wait a little bit longer when needed,
 	// making this test less flaky due to delays in the deployment.
-	_, err := utils.HTTPGetNTimes(externalURL, numHealthChecks)
+	_, err = utils.HTTPGetNTimes(externalURL, numHealthChecks)
 	require.NoError(t, err)
 
+	t.Logf("externalURL is '%s'\n", externalURL)
+
 	for _, tt := range serviceinvocationTests {
 		t.Run(tt.in, func(t *testing.T) {
 			body, err := json.Marshal(testCommandRequest{
 				RemoteApp: tt.remoteApp,
-				Method:    tt.method,
+				Method:    tt.appMethod,
+			})
+			require.NoError(t, err)
+
+			resp, err := utils.HTTPPost(
+				fmt.Sprintf("%s/tests/invoke_test", externalURL), body)
```

Comment:
Why you don't need the appMethod here?

Reply:
appMethod is the method to call via dapr service invokcation.  /tests/invoke_tests, is a non-dapr endpoint used just to talk to the app.  That handler body which contains appMethod then calls appMethod via dapr from there.